packages/astro/src/integrations/hooks.ts

```diff
@@ -151,6 +155,9 @@ export async function runHookConfigSetup({
 	for (let i = 0; i < updatedConfig.integrations.length; i++) {
 		const integration = updatedConfig.integrations[i];
 
+		const codegenDir = new URL(normalizeCodegenDir(integration.name), settings.dotAstroDir);
+		await fs.promises.mkdir(codegenDir, { recursive: true });
```

Comment:
Does this mean we're making a directory for every integration even if they are not using it? Assuming so, let's talk about that. It feels like a lot initially, I could be overthinking this though.

An alternative would be something like a `createCodegenDir()` that returns the path, so the integration explicitly asks for it.

Reply:
That sounds good to me